### PR TITLE
fix `ImportCommunityPopup` issues, remove private key importing

### PR DESCRIPTION
**Pull request by igor-sirotin**
Fixes https://github.com/status-im/status-desktop/issues/12465
Fixes https://github.com/status-im/status-desktop/issues/12542
Drops the usage of `parseCommunitySharedUrl` (https://github.com/status-im/status-desktop/issues/12356)

### What does the PR do

1. Correctly show result of `requestCommunityInfo` call
2. Remove QML timeout, trust status-go timeout instead
3. Remove support for private key importing
4. Remove automatic call for `reuestCommunityInfo` from `getCommunityDetails`. 
Because the former returns nothing and requires binding to nim signals.
5. Update status-go to include https://github.com/status-im/status-go/pull/4194
6. storybook page for `ImportCommunityPopup`

### Affected areas

ImportCommunityPopup

### Screenshot of functionality (including design for comparison)

https://github.com/status-im/status-desktop/assets/25482501/acc37804-a043-4652-abfc-25ed68f2c286

https://github.com/status-im/status-desktop/assets/25482501/28e61e4a-5c6d-4221-a881-32afeb1f2ba7

### CODE REVIEW

File: ui/app/AppLayouts/Communities/stores/CommunitiesStore.qml [hunk 1]
```diff
@@ -251,5 +248,9 @@ QtObject {
         function onCommunityInfoAlreadyRequested() {
           root.communityInfoAlreadyRequested()
         }
+
+        function onCommunityInfoRequestCompleted(communittyId, erorrMsg) {
+            root.communityInfoRequestCompleted(communittyId, erorrMsg)
```
Comment (jrainville) [inline ui/app/AppLayouts/Communities/stores/CommunitiesStore.qml:253]:
```suggestion
        function onCommunityInfoRequestCompleted(communityId, erorrMsg) {
            root.communityInfoRequestCompleted(communityId, erorrMsg)
```

**Review by jrainville — APPROVED** [review]

File: ui/app/AppLayouts/Communities/stores/CommunitiesStore.qml (hunk 1 above)
Comment (caybro) [reply ui/app/AppLayouts/Communities/stores/CommunitiesStore.qml]: You might want to fix the `erorrMsg` typo too :laughing:

Comment (jrainville) [reply ui/app/AppLayouts/Communities/stores/CommunitiesStore.qml]: Damn, the community typo was so egregious I missed the second typo :joy:

File: storybook/main.qml [hunk 2]
```diff
@@ -140,6 +140,20 @@ ApplicationWindow {
                         onClicked: settingsPopup.open()
                     }
 
+                    CheckBox {
+                        id: windowAlwaysOnTopCheckBox
+
+                        Layout.fillWidth: true
+
+                        text: "Always on top"
```
Comment (caybro) [inline storybook/main.qml:148]: Um, why is this needed? :thinking:

File: ui/imports/shared/popups/ImportCommunityPopup.qml [hunk 3]
```diff
@@ -15,111 +15,103 @@ import StatusQ.Controls 0.1
...
-            loading = true
-            const key = isPublicKey ? Utils.getCompressedPk(publicKey) :
-                                      root.store.getCommunityPublicKeyFromPrivateKey(inputKey, true /*importing*/);
-
-            const details = root.store.getCommunityDetails(key)
-            if (!!details) // the above can return `null` in which case we continue loading
-                loading = false
-            return details
-        }
+        property bool communityFound: !!d.communityDetails && !!d.communityDetails.name
```
Comment (caybro) [inline ui/imports/shared/popups/ImportCommunityPopup.qml:31]: `readonly`?

File: ui/imports/shared/popups/ImportCommunityPopup.qml [hunk 4]
```diff
@@ -15,111 +15,103 @@ import StatusQ.Controls 0.1
...
-                if (!Utils.isCommunityPublicKey(key))
-                    return ""
-                if (!Utils.isCompressedPubKey(key))
-                    return key
-                return Utils.changeCommunityKeyCompression(key)
-            } else {
-                return Utils.getCommunityDataFromSharedLink(inputKey).communityId;
+            if (Utils.isStatusDeepLink(inputKey)) {
+                const linkData = Utils.getCommunityDataFromSharedLink(inputKey)
+                return !linkData ? "" : linkData.communityId
```
Comment (caybro) [inline ui/imports/shared/popups/ImportCommunityPopup.qml:42]:
```suggestion
                return !!linkData ? linkData.communityId : ""
```

I always find this double negative idiom ugly :)

File: ui/imports/shared/popups/ImportCommunityPopup.qml [hunk 5]
```diff
@@ -15,111 +15,103 @@ import StatusQ.Controls 0.1
...
+        }
+
+        onPublicKeyChanged: {
+            // call later to make sure all proeprties used by `updateCommunityDetails` are udpated
+            Qt.callLater(() => { d.updateCommunityDetails(true) })
         }
-        readonly property bool isInputValid: isPrivateKey || isPublicKey
     }
 
     Timer {
```
Comment (caybro) [inline ui/imports/shared/popups/ImportCommunityPopup.qml:82]: Uhm, remove the `Timer` when it's not used?

File: ui/imports/shared/popups/ImportCommunityPopup.qml [hunk 6]
```diff
@@ -15,111 +15,103 @@ import StatusQ.Controls 0.1
...
-                d.communityDetails = null
-                d.importErrorMessage = msg
+        function onCommunityInfoRequestCompleted(communityId, errorMsg) {
+            if (!d.communityInfoRequested)
+                return
+
+            d.communityInfoRequested = false
+
+            if (errorMsg !== "") {
+                d.importErrorMessage = "Couldn't find community"
```
Comment (caybro) [inline ui/imports/shared/popups/ImportCommunityPopup.qml:98]:
```suggestion
                d.importErrorMessage = qsTr("Couldn't find community")
```

**Review by caybro — COMMENTED** [review]
Overall LGTM, just some minor points

File: storybook/main.qml (hunk 2 above)
Comment (igor-sirotin) [reply storybook/main.qml:148]: Well, sometimes it's good to have it on top while you're live-coding.
I saw @micieslak using this option time to time, but he uses OS instruments to do this. And MacOS doesn't have such feature 🤷‍♂️

File: ui/imports/shared/popups/ImportCommunityPopup.qml (hunk 4 above)
Comment (igor-sirotin) [reply ui/imports/shared/popups/ImportCommunityPopup.qml:42]: I don't get it, you find it ugly, but still suggest? 😄

File: ui/imports/shared/popups/ImportCommunityPopup.qml (hunk 5 above)
Comment (igor-sirotin) [reply ui/imports/shared/popups/ImportCommunityPopup.qml:82]: oh, totally missed to remove!

**Review by caybro — APPROVED** [review]
LGTM
